tc/core/polyhedral/cuda/memory_promotion_heuristic.cc

```diff
@@ -436,14 +436,25 @@ bool isPromotableToRegistersBelow(
 }
 
 /*
- * Starting from the root, find bands where depth is reached.  Using
+ * Starting from the root, find bands where depth is reached.  If zero depth is
+ * requested, insert a zero-dimensional band node below the root (or the
+ * context node if present) and return it.  Otherwise, use
  * DFSPreorder to make sure order is specified and consistent for tests.
  */
 std::vector<detail::ScheduleTree*> bandsContainingScheduleDepth(
     detail::ScheduleTree* root,
     size_t depth) {
   using namespace tc::polyhedral::detail;
 
+  if (depth == 0) {
+    auto node = root;
+    if (root->numChildren() > 0 &&
+        root->child({0})->elemAs<detail::ScheduleTreeElemContext>()) {
+      node = root->child({0});
+    }
+    return {insertNodeBelow(node, ScheduleTree::makeEmptyBand(root))};
```

Comment:
This is a bit distracting.  Can we extract some or all of this?
Maybe a `skipContext` or a `insertTopLevelEmptyBand`